modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java

```diff
@@ -319,8 +319,18 @@ private Price _getPrice(
 		Price price = new Price() {
 			{
 				currency = commerceCurrency.getName(locale);
-				price = unitPrice.doubleValue();
 				priceFormatted = unitPriceCommerceMoney.format(locale);
+				priceOnApplication =
+					commerceProductPrice.isPriceOnApplication();
+
+				setPrice(
+					() -> {
+						if (unitPrice != null) {
```

Comment:
Please check if this is still needed unitPrice should be 0 not null

Reply:
Not needed anymore for [this](https://github.com/liferay-commerce/liferay-portal/blob/bafacf008add38955f7c1f00eb3ef43ac30f2a8b/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/price/BaseCommerceProductPriceCalculation.java#L629)